src/code-templates/libraries/logger/index.ts

```diff
@@ -1,9 +1,16 @@
+import { context } from '@practica/async-local-storage';
 import { Logger, LoggerConfiguration } from './definition';
 import PinoLogger from './pino.logger';
 
 export class LoggerWrapper implements Logger {
   #underlyingLogger: Logger | null = null;
 
+  get #configuredUnderlyingLogger(): Logger {
```

Comment:
Design dilemma 1 - Getter or function? Not everyone are familiar with getters, functions are the simplest language tool and should be the default IMO. It's also not the typical getter case where you want to wrap an inner field with some calculated property rather a case some work/effect is needed to make it happen (IO work might occur during this phase)

Design dilemma 2 - Whether property or function, the name can't be 'underlyingLogger' because they don't represent the same thing. This property represents a logger that is already configured and ready to work, call it accordingly? getInitializeLogger/initializedLogger?

Reply:
I agree with your thoughts on the 1st dilemma and changed it to a function
and in the 2nd dilemma changed it to `getInitializeLogger`

Anyway, I don't think that we should try to reconfigure every time or allow resetting, why? because it adds an additional unnecessary function before each log IMO...